provider/src/main/kotlin/org/gradle/kotlin/dsl/KotlinInitScript.kt

```diff
@@ -0,0 +1,95 @@
+    open fun initscript(@Suppress("unused_parameter") block: ScriptHandlerScope.() -> Unit) = Unit
+
+    /**
+     * Creates a [ConfigurableFileCollection] containing the given files.
+     *
+     * You can pass any of the following types to this method:
+     *
+     * - A [CharSequence], including [String] as defined by [KotlinSettingsScript.file].
+     * - A [File] as defined by [KotlinSettingsScript.file].
+     * - A [java.nio.file.Path] as defined by [KotlinSettingsScript.file].
+     * - A [URI] or [java.net.URL] as defined by [KotlinSettingsScript.file].
+     * - A [org.gradle.api.file.Directory] or [org.gradle.api.file.RegularFile]
+     *   as defined by [KotlinSettingsScript.file].
+     * - A [Sequence], [Array] or [Iterable] that contains objects of any supported type.
+     *   The elements of the collection are recursively converted to files.
+     * - A [org.gradle.api.file.FileCollection].
+     *   The contents of the collection are included in the returned collection.
+     * - A [org.gradle.api.provider.Provider] of any supported type.
+     *   The provider's value is recursively converted to files. If the provider represents an output of a task,
+     *   that task is executed if the file collection is used as an input to another task.
+     * - A [java.util.concurrent.Callable] that returns any supported type.
+     *   The callable's return value is recursively converted to files.
+     *   A `null` return value is treated as an empty collection.
+     * - A [org.gradle.api.Task].
+     *   Converted to the task's output files.
+     *   The task is executed if the file collection is used as an input to another task.
+     * - A [org.gradle.api.tasks.TaskOutputs].
+     *   Converted to the output files the related task.
+     *   The task is executed if the file collection is used as an input to another task.
+     * - Anything else is treated as a failure.
+     *
+     * The returned file collection is lazy, so that the paths are evaluated only when the contents of the file
+     * collection are queried. The file collection is also live, so that it evaluates the above each time the contents
+     * of the collection is queried.
+     *
+     * The returned file collection maintains the iteration order of the supplied paths.
+     *
+     * The returned file collection maintains the details of the tasks that produce the files,
+     * so that these tasks are executed if this file collection is used as an input to some task.
+     *
+     * This method can also be used to create an empty collection, which can later be mutated to add elements.
+     *
+     * @param paths The paths to the files. May be empty.
+     * @return The file collection.
+     */
+    @Suppress("unused")
+    fun files(vararg paths: Any): ConfigurableFileCollection =
+        fileOperations.files(paths)
+
+    private
+    val fileOperations by lazy {
+        fileOperationsFor(gradle, gradle.startParameter.currentDir)
```

Comment:
Is this the current directory of the `init.gradle.kts` file or the directory where gradle is invoked from?

Reply:
:x: To be on par with the Groovy DSL implementation it should be:
- the directory of the init script if it is a regular file
- the base dir of the default `FileResolver` if the init script is not a regular file (e.g. remote URI)

See https://github.com/gradle/gradle/blob/010abecca658c2d9c92c2d223b8a7554eff0ccf4/subprojects/core/src/main/java/org/gradle/groovy/scripts/DefaultScript.java#L88-L93

Reply:
So what you're saying is that this is incorrect as it is?